### PR TITLE
feat: `dm_from_con(learn_keys = TRUE)` works for MariaDB

### LINKED ISSUE #366: dm_learn_from_db and MySQL
looking forward to playing with this within MySQL...

### LINKED ISSUE #1124: MariaDB supports `dm_from_con(learn_keys = TRUE)`
after #1123. Need to adapt documentation to that effect. Search for `dm_from_con` in vignettes.

### PR DESCRIPTION
**Pull request by krlmlr**
for relational.fit .

Closes #366. Closes #1124.

### CODE REVIEW

**Comment by krlmlr** [comment]
@maelle: Please feel free to add to this branch.

**Comment by maelle** [comment]
Noting that I do not get any failure locally but I do remember the text about MariaDB needs to be updated anyway.

**Comment by maelle** [comment]
Oh, actually I need to try `lazy = FALSE`.

**Comment by maelle** [comment]
@krlmlr in the dm.Rmd vignette, do you want to

- remove mentions of how to add primary keys
- change the example DB to be able to demonstrate how to add primary keys?

File: R/db-helpers.R [hunk 1]
```diff
@@ -91,7 +91,7 @@ repair_table_names_for_db <- function(table_names, temporary, con, schema = NULL
     names <- unique_db_table_name(names)
   } else {
     # permanent tables
-    if (!is.null(schema) && !is_mssql(con) && !is_postgres(con)) {
+    if (!is.null(schema) && !is_mssql(con) && !is_postgres(con) && !is_mariadb(con)) {
```
[the remark this answers is not in the document]
Comment (maelle) [reply R/db-helpers.R:94]: Done in #1127

**Comment by maelle** [comment]
@krlmlr should the tests that have `skip_if_src_not(c("mssql", "postgres"))` have been updated or do they exclude MariaDB for other reasons?

It might be nice to replace `skip_if_src_not(c("mssql", "postgres"))` with something more self-explanatory?

**Pull request merged**